src/webui/webapplication.cpp

```diff
@@ -306,9 +308,31 @@ void WebApplication::action_command_shutdown()
 void WebApplication::action_command_download()
 {
     CHECK_URI(0);
-    CHECK_PARAMETERS("urls");
+    CHECK_PARAMETERS("urls" << "savepath" << "label" << "cookie");
```

Comment:
Whit this checks the 4 parameters are compulsory, so you are breaking backwards compatibility. This API is used by 3rd party apps. I prefer to make "urls" compulsory as before and the other parameters optional.
if not "savepath" set the default value
if not "label" set no label
if noy "cookie" set no cookie